application/comit_node/src/http_api/rfc003/handlers/get_swaps.rs

```diff
@@ -73,13 +73,11 @@ pub fn handle_get_swaps<T: MetadataStore<SwapId>, S: StateStore>(
                             parameters,
                         };
 
-                        let mut hal_resource = HalResource::new(swap);
-                        hal_resource.with_link("self", swap_path(id));
-
-                        for action in actions {
-                            let link = new_action_link(&id, action);
-                            hal_resource.with_link(action, link);
-                        }
+                        let hal_resource = HalResource::new(swap).with_link("self", swap_path(id));
+                        let hal_resource = actions.iter().fold(hal_resource, |acc, action| {
```

Comment:
Can you take ownership of `actions` here? Then you can do `into_iter` and avoid the deref of action in line 78.